### PR TITLE
Create Devcontainer build

### PR DESCRIPTION
**Pull request by bennothommo**
This PR introduces the ability for Winter to run in a [development container](https://containers.dev/), which is basically a pre-constructed configuration for Docker that creates a fully-functional development environment for Winter. It is supported by VSCode, PHPStorm and many other editors to easily create local development copies, and also allows us to use GitHub Codespaces to test PRs or make edits to the core directly in GitHub.

A more thorough explanation can be found in the [README](https://github.com/wintercms/winter/blob/c53e2337684a66b158e65bce6c4efcdfd79cc9e2/.devcontainer/README.md) for the dev container configuration.

### CODE REVIEW

**Comment by bennothommo** [comment]
Here's how it looks in GitHub Codespaces:

screenshot

File: .devcontainer/local-features/bootstrap-winter/bootstrap.sh [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+git restore config
+
+cp ${PWD}/.devcontainer/.vscode/launch.json ${PWD}/.vscode/launch.json
+
+if [ "${CODESPACES}" = "true" ]; then
+    echo "### Configure for Codespaces"
+    php ${PWD}/.devcontainer/local-features/bootstrap-winter/codespaces.php
+    git update-index --assume-unchanged config/app.php
+    gh codespace ports visibility 8080:public -c $CODESPACE_NAME
+fi
```
Comment (LukeTowers) [inline .devcontainer/local-features/bootstrap-winter/bootstrap.sh]:
```suggestion
fi

```

**Comment by LukeTowers** [comment]
@jaxwilko do you have any input?

@bennothommo does the .devcontainer folder get included when creating winter projects with composer?

**Comment by bennothommo** [comment]
@LukeTowers currently, yes it would. I'm in two trains of thought on whether to keep it for Composer installs - on the one hand, using Composer would imply they have a working dev environment (or at least PHP). On the other hand, it's still quite useful for getting a project off the ground quickly without having to bootstrap it.

It might be worth stripping it out for installs through the Web Installer though.

File: .devcontainer/local-features/bootstrap-winter/codespaces.php [hunk 2]
```diff
@@ -0,0 +1,25 @@
+<?php
+
+$root = dirname(dirname(dirname(__DIR__)));
```
Comment (jaxwilko) [inline .devcontainer/local-features/bootstrap-winter/codespaces.php]:
```suggestion
$root = dirname(__DIR__, 3);
```

File: .devcontainer/local-features/bootstrap-winter/update-composer.php [hunk 3]
```diff
@@ -0,0 +1,35 @@
+<?php
+$composer = json_decode(file_get_contents(dirname(dirname(dirname(__DIR__))) . '/composer.json'), true);
```
Comment (jaxwilko) [inline .devcontainer/local-features/bootstrap-winter/update-composer.php]:
```suggestion
$composerPath = dirname(__DIR__, 3) . '/composer.json';
$composer = json_decode(file_get_contents($composerPath), true);
```

File: .devcontainer/local-features/bootstrap-winter/update-composer.php [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+
+// Install Workshop theme
+if (
+    !in_array('winter/wn-workshop-theme', array_keys($composer['require']))
+) {
+    $composer['require']['winter/wn-workshop-theme'] = 'dev-main';
+}
+
+file_put_contents(
+    dirname(dirname(dirname(__DIR__))) . '/composer.json',
```
Comment (jaxwilko) [inline .devcontainer/local-features/bootstrap-winter/update-composer.php]:
```suggestion
    $composerPath,
```

File: .devcontainer/local-features/bootstrap-winter/update-composer.php [hunk 5]
```diff
@@ -0,0 +1,35 @@
...
+) {
+    $composer['require']['winter/wn-pages-plugin'] = 'dev-main';
+}
+
+// Install Workshop theme
+if (
+    !in_array('winter/wn-workshop-theme', array_keys($composer['require']))
+) {
+    $composer['require']['winter/wn-workshop-theme'] = 'dev-main';
+}
```
Comment (jaxwilko) [inline .devcontainer/local-features/bootstrap-winter/update-composer.php]:
```suggestion
$packages = [
    'winter/wn-test-plugin' => 'dev-main',
    'winter/wn-blog-plugin' => 'dev-main',
    'winter/wn-blog-plugin' => 'dev-main',
    'winter/wn-workshop-theme' => 'dev-main',
];

// Install Winter packages
foreach ($packages as $package => $version) {
    if (!in_array($package, array_keys($composer['require']))) {
        $composer['require'][$package] = $version;
    }
}
```

File: .devcontainer/README.md [hunk 6]
```diff
@@ -0,0 +1,62 @@
...
+- NodeJS 22 (including `npm`)
+- Git
+
+## Using in your own projects
+
+You may use this development environment for your own projects, making it a great starting point to hit the ground running with Winter. It is recommended that you *disable* the `bootstrap-winter` feature when using this environment for your own projects.
+
+You may disable this feature by modifying the `.devcontainer/devcontainer.json` file before running the container and commenting out the feature:
+
+```json
```
Comment (jaxwilko) [inline .devcontainer/README.md]: To fix parsing of comments in json
```suggestion
```json5
```

**Comment by jaxwilko** [comment]
@LukeTowers Looks cool to me, I generally just spin up my own containers and probably wouldn't use the one provided here but I can see the value for some people.

@bennothommo I would lean more to excluding it when installed with composer, generally if somebody has php install then they probably have it set up... or can run the php built-in web server. Perhaps it should only be downloaded when cloning the repo (i.e. not in the web installer or via composer).

File: .devcontainer/local-features/bootstrap-winter/codespaces.php (hunk 2 above)
Comment (bennothommo) [reply .devcontainer/local-features/bootstrap-winter/codespaces.php]: Didn't even know that was a thing - nice!

**Comment by bennothommo** [comment]
I'll merge this now as it's a developmental change that won't affect anyone else. We can tweak if we need to, down the line.

**Pull request merged**